src/teleport.cpp

```diff
@@ -168,7 +168,7 @@ bool teleport::teleport_to_point( Creature &critter, tripoint target, bool safe,
     }
     critter.setpos( target );
     if( collision ) {
-        g->fling_creature( critter, units::from_degrees( collision_angle - 180 ), 50 );
+        g->fling_creature( get_creature_tracker().creature_at<Creature>( target ), units::from_degrees( collision_angle - 180 ), 50 );
```

Comment:
```suggestion
        &critter, units::from_degrees( collision_angle - 180 ), 50 );
```
More or less a wild guess from doing a find in files 'fling_creature', but does this do the trick?